scripts/main.js

```diff
@@ -388,58 +472,133 @@
+        }
+
+        // Shield (true / false)
+        message.shield = user.has_shield;
+
+        // Total Power, Luck, Attraction
+        message.total_power = user.trap_power;
+        if (user_post.trap_power !== user.trap_power) {
+            debug_logs.push(`User setup power changed from ${user.trap_power} to ${user_post.trap_power}`);
+        }
+
+        message.total_luck = user.trap_luck;
+        if (user_post.trap_luck !== user.trap_luck) {
+            debug_logs.push(`User setup luck changed from ${user.trap_luck} to ${user_post.trap_luck}`);
+        }
+
+        message.attraction_bonus = Math.round(user.trap_attraction_bonus * 100);
+        if (user_post.trap_attraction_bonus !== user.trap_attraction_bonus) {
+            debug_logs.push(`User setup attraction bonus changed from ${user.trap_attraction_bonus} to ${user_post.trap_attraction_bonus}`);
+        }
 
         // Setup components
-        let components = [
+        const components = [
             { prop: 'weapon', message_field: 'trap', required: true, replacer: /\ trap/i },
             { prop: 'base', message_field: 'base', required: true, replacer: /\ base/i },
-            { prop: 'trinket', message_field: 'charm', required: false, replacer: /\ charm/i },
-            { prop: 'bait', message_field: 'cheese', required: true, replacer: /\ cheese/i }
+            { prop: 'bait', message_field: 'cheese', required: true, replacer: /\ cheese/i },
+            { prop: 'trinket', message_field: 'charm', required: false, replacer: /\ charm/i }
         ];
-        // Some components are required.
-        let missing = components.filter(component => component.required === true && !user_resp.hasOwnProperty(component.prop + '_name'));
+        // All pre-hunt users must have a weapon, base, and cheese.
+        const missing = components.filter(component => component.required === true && !user.hasOwnProperty(`${component.prop}_name`));
         if (missing.length) {
-            console.log('MH Helper: Missing required setup component:' + missing.map(c => c.message_field).join(', '));
-            return "";
+            window.console.error(`MH Helper: Missing required setup component: ${missing.map(c => c.message_field).join(', ')}`);
+            return null;
         }
         // Assign component values to the message.
         components.forEach(component => {
-            let prop_name = component.prop + '_name';
-            let prop_id = component.prop + '_item_id';
-            if (!user_resp[prop_name]) return;
-            message[component.message_field] = {
-                id: user_resp[prop_id],
-                name: user_resp[prop_name].replace(component.replacer, '')
+            const prop_name = `${component.prop}_name`;
+            const prop_id = `${component.prop}_item_id`;
+            const item_name = user[prop_name];
+            message[component.message_field] = (!item_name) ? {} : {
```

Comment:
Note: For truly charmless hunts, this will create a field `charm: {}`

I'm not sure how that field is handled by Jack's server; if it's problematic this line should be updated

Reply:
good note, thank you. I believe i will catch it just fine: https://github.com/DevJackSmith/mh-hunt-helper/blob/36d51a8a8ae6a9596119ee79c61079e880e241b1/Server/intake.php#L111 But i'd keep it the same type in future if possible: if the value is usually a string, then "" if it's missing. If the value is usually and object, then {} if it's missing.